src/core/Observer.ts

```diff
@@ -336,36 +399,123 @@ export class Observer {
     return rtDocument;
   }
 
+  /**
+   * Start subscription or pulling on the collection
+   *
+   * @internal
+   */
+  watchCollection (index: string, collection: string) {
+    return this.mode === 'realtime'
+      ? this.resubscribe(index, collection)
+      : this.restartPulling();
+  }
+
+  private restartPulling (): Promise<void> {
+    return this.clearPullingTimer()
+      .then(() => {
+        if (this.documentsByCollection.size !== 0) {
+          this.pullingTimer = setInterval(
+            this.pullingHandler.bind(this),
+            this.options.pullingDelay);
+        }
+      });
+  }
+
+  /**
+   * Use the document.mGet method to pull documents from observed collections
+   * and update internal realtime documents.
+   *
+   * This method never returns a rejected promise.
+   */
+  private pullingHandler (): Promise<void> {
+    const promises = [];
+
+    for (const observedDocuments of this.documentsByCollection.values()) {
+      const promise = this.sdk.document.mGet(
+        observedDocuments.index,
+        observedDocuments.collection,
+        observedDocuments.ids
+      )
+        .then(({ successes, errors }) => {
+          for (const document of successes) {
+            const urn = documentUrn(
+              observedDocuments.index,
+              observedDocuments.collection,
+              document._id);
+
+            const rtDocument = this.documents.get(urn);
+            Object.assign(rtDocument._source, document._source);
+          }
+
+          for (const deletedDocumentId of errors) {
```

Comment:
Are we sure that `errors` are forcibly deleted documents?

Reply:
Yes, in this API action `errors` is an array of not found document ids